audio_classification/android/app/src/main/java/com/google/edgeai/examples/audio_classification/AudioClassificationHelper.kt

```diff
@@ -0,0 +1,278 @@
+    }
+
+    suspend fun setupInterpreter() {
+        interpreter = try {
+            val tfliteBuffer = FileUtil.loadMappedFile(context, options.currentModel.fileName)
+            Log.i(TAG, "Done creating TFLite buffer from ${options.currentModel}")
+            labels = getModelMetadata(tfliteBuffer)
+            Interpreter(tfliteBuffer, Interpreter.Options().apply {
+                numThreads = options.threadCount
+                useNNAPI = options.delegate == Delegate.NNAPI
+            })
+        } catch (e: IOException) {
+            throw IOException("Failed to load TFLite model - ${e.message}")
+        } catch (e: Exception) {
+            throw Exception("Failed to create Interpreter - ${e.message}")
+        }
+    }
+
+    /*
+    * Starts sound classification, which triggers running on IO Thread
+    */
+    @SuppressLint("MissingPermission")
+    suspend fun startRecord() {
+        withContext(Dispatchers.IO) {
+            // Inspect input and output specs.
+            val inputShape = interpreter?.getInputTensor(0)?.shape() ?: return@withContext
+
+            /**
+             * YAMNET input: float32[15600]
+             * Speech Command input: float32[1,44032]
+             */
+            val modelInputLength =
+                inputShape[if (options.currentModel == TFLiteModel.YAMNET) 0 else 1]
+            audioManager = AudioManager(modelInputLength, options.overlapFactor)
+
+            previousAudioArray = FloatArray(0)
+            audioManager!!.record().collect {
+                val array = convertShortToFloat(it)
+                startRecognition(array)
+            }
+        }
+    }
+
+    private suspend fun startRecognition(audioArray: FloatArray) {
+        // Inspect input and output specs.
+        val inputShape = interpreter?.getInputTensor(0)?.shape() ?: return
+
+        /**
+         * YAMNET input: float32[15600]
+         * Speech Command input: float32[1,44032]
+         */
+        val modelInputLength = inputShape[if (options.currentModel == TFLiteModel.YAMNET) 0 else 1]
+
+        val outputShape = interpreter?.getOutputTensor(0)?.shape() ?: return
+        val modelNumClasses = outputShape[1]
+        // Fill the array with NaNs initially.
+        val predictionProbs = FloatArray(modelNumClasses) { Float.NaN }
+
+        val inputBuffer = FloatBuffer.allocate(modelInputLength)
+
+        coroutineScope {
+            if (modelInputLength <= 0) {
+                Log.e(TAG, "Switches: Cannot start recognition because model is unavailable.")
+                return@coroutineScope
+            }
+            val outputBuffer = FloatBuffer.allocate(modelNumClasses)
+            // Put audio data to buffer
+            if (previousAudioArray != null) {
+                // Put previous audio array first
+                inputBuffer.put(previousAudioArray)
+                // Case: overlap > 50%
+                if (audioArray.size < modelInputLength - previousAudioArray!!.size) {
```

Comment:
Overlap shouldn't be adding into the same buffer. It should be that there are two separate streams that start their continual classification at overlap_percentage*length_of_expected_audio_clip time difference. Let's just get rid of overlap all together for these examples to make things easier.

Reply:
@PaulTR I tried it with two separate streams but the results are a bit inaccurate, it seems I made a mistake.
First I will remove the overlap in this PR first.
Then I will update it later in another PR